src/watch.rs

```diff
@@ -95,8 +95,11 @@ impl WatchSystem {
     }
 
     fn update_ignore_list(&mut self, path: PathBuf) {
-        if !self.ignores.contains(&path) {
-            self.ignores.push(path);
+        let canon_path = path.canonicalize().expect(
+            format!("Could not canonicalize path: {}", path.to_string_lossy()).as_ref()
+        );
```

Comment:
Similar to @philip-peterson's excellent recommendations elsewhere, let's not use panicking methods. So instead of expect here, you could do:
```suggestion
        let canon_path = path.canonicalize().with_context(||
            format!("Could not canonicalize path: {}", path.to_string_lossy()).as_ref()
        );
```

You'll have to update this methods signature to return a result as well though. Just a simple `-> Result<()> {` at the end of the function signature should do the trick.